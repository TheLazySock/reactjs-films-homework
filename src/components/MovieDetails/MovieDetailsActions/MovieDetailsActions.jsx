import React from 'react';
import PropTypes from 'prop-types';
import style from './MovieDetailsActions.scss';

import WatchNowButton from './WatchNowButton';
import ViewInfo from './ViewInfo';

class MovieDetailsActions extends React.Component {
  render() {
    const { videoUrl, info } = this.props;

    return (
      <div className={style.movieDetailsActions}>
        <WatchNowButton videoUrl={videoUrl} />
        <ViewInfo onClick={this.handleShowInfo} info={info} />
      </div>
    );
  }
}

MovieDetailsActions.propTypes = {
  videoUrl: PropTypes.string.isRequired,
  info: PropTypes.string.isRequired,
};

export default MovieDetailsActions;
