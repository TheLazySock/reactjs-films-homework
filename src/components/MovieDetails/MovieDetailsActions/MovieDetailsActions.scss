.movieDetailsActions {
  position: relative;
  color: white;
  display: flex;
}
