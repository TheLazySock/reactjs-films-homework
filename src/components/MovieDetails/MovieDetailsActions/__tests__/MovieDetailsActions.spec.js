import React from 'react';
import renderer from 'react-test-renderer';

import MovieDetailsActions from '../index';

describe('MovieDetailsActions', () => {
  it('renders correctly', () => {
    const output = renderer
      .create(<MovieDetailsActions videoUrl="https://" info="info" />)
      .toJSON();
    expect(output).toMatchSnapshot();
  });

  it('show description on button click', () => {
    const output = renderer.create(<MovieDetailsActions videoUrl="https://" info="info" />);
    output.root.findByProps({ className: 'viewInfoBtn' }).props.onClick();
    expect(output).toMatchSnapshot();
  });
});
