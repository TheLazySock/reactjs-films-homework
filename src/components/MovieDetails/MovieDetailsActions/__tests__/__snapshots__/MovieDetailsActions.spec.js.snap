// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsActions renders correctly 1`] = `
<div
  className="movieDetailsActions"
>
  <button
    className="watchVideoBtn"
    type="button"
  >
    Watch Now
  </button>
  <div
    className="viewInfoContainer"
  >
    <button
      className="viewInfoBtn"
      onClick={[Function]}
      type="button"
    >
      View Info
    </button>
  </div>
</div>
`;

exports[`MovieDetailsActions show description on button click 1`] = `
<div
  className="movieDetailsActions"
>
  <button
    className="watchVideoBtn"
    type="button"
  >
    Watch Now
  </button>
  <div
    className="viewInfoContainer"
  >
    <div>
      <p
        className="modalInfo"
      >
        info
      </p>
    </div>
    <button
      className="viewInfoBtn"
      onClick={[Function]}
      type="button"
    >
      View Info
    </button>
  </div>
</div>
`;
